### DISCUSSION TITLE
How do I properly work with Allure decorators?

### DISCUSSION THREAD

**Discussion by SergeyGuschin** [seleniumbase/SeleniumBase#1811] [General]
Hello Michael. Thank you again for the great work you've done. I have a question about allure decorators. How can I properly do them in scripts? When I try to generate and run the allure report I have zero cases in it. Thank you.
```
class LoginToSite(LoginPage):
    @allure.title('Login to webapp')
    def test_login_to_site(self):
        with allure.step("steps to login"):
            self.login_to_site()
            time.sleep(2)
```
![image](https://user-images.githubusercontent.com/96178944/226214401-6e373a35-54ee-4949-9e2c-230d0d8f544e.png)

**Comment by mdmintz** [comment]
For running SeleniumBase tests with Allure Reports:

1. ``pip install allure-pytest``
2. Add ``--alluredir=allure_results`` to a ``pytest`` run command.
3. Display the report: ``allure serve allure_results/``

Here's how that report looks for ``pytest coffee_cart_tests.py --alluredir=allure_results --rs``
(On https://github.com/seleniumbase/SeleniumBase/blob/master/examples/coffee_cart_tests.py)

<img width="1280" alt="Screenshot 2023-03-19 at 7 05 32 PM" src="https://user-images.githubusercontent.com/6788579/226215447-5e6b8601-e239-46ab-a39c-0b17d643a2ff.png">

--------

See https://docs.qameta.io/allure/#_pytest for more info on using Allure with pytest. As for Allure decorators, you should check with them about it, given that it is off-topic and not part of SeleniumBase directly.

**Comment by SergeyGuschin** [comment]
Thanks again for your quick help. I used to do everything you suggested in the answer. But my problem was that I opened the report with the wrong command ))))
`allure serve /allure_results`
